### ISSUE TITLE
Add Data.Constraints.Char for GHCs where it is available.

### ISSUE THREAD

**Issue by phadej** [ekmett/constraints#117]

**Comment by RyanGlScott** [comment]
Do you have suggestions for what operations such as module should provide? One of the main benefits of `Data.Constraints.{Nat,Symbol}` is that they provide operations that aren't expressible with the API in `GHC.TypeLits`. I'm not sure off the top of my head which operations would be most useful for `Data.Constraints.Char` to provide.

**Comment by phadej** [comment]
```haskell
unconsSymbol :: KnownSymbol sym :- KnownMaybe (KnownPair KnownChar KnownSymbol) (UnconsSymbol sym)
consSymbol :: ..
```

Though the problem is where to put `KnownMaybe` and `KnownPair`, and what they should look like.

But at least

```haskell
charToNat :: KnownChar c :- KnownNat (CharToNat c)
```

is not problematic, and probably `natToChar` with additional constraint.

**Comment by RyanGlScott** [comment]
I've implemented this in #122.
